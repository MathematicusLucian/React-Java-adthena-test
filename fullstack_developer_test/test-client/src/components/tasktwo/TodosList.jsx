import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

// This is a functional component

const TodosList = ({ todos }) => {
  
  return (
    <div className="todoList">
        {todos
            .map((t) => (
                <div className="todoItem" key={t.id}>
                    {t.id}: {t.title}
                </div>
            ))              

        }
    </div>)
};

TodosList.propTypes = {
    todos: PropTypes.any,
};

TodosList.defaultProps = {
    message: 'Adthena',
};

export default TodosList;
