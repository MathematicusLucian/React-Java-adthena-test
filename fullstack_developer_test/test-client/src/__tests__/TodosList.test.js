import React from 'react';
import { shallow, render, mount } from "enzyme";
import TodosList from './../components/tasktwo/TodosList';

describe('TodosList', () => {

  let wrapper;
  const todosStub = [
  {
      "userId": 1,
      "id": 1,
      "title": "delectus aut autem",
      "completed": false
  },
  {
      "userId": 1,
      "id": 2,
      "title": "quis ut nam facilis et officia qui",
      "completed": false
  },
  {
      "userId": 1,
      "id": 3,
      "title": "fugiat veniam minus",
      "completed": false
  },
  {
      "userId": 1,
      "id": 4,
      "title": "et porro tempora",
      "completed": true
  }];

  beforeEach(() => {
    const props = {};
    wrapper = shallow(<TodosList {...props} todos={todosStub}></ TodosList>);
  });

  it('should render correctly', () => {
    shallow(<TodosList todos={todosStub} />);
  });

  it('should render - shallow', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should render (snapshot)', () => {
    const rendered = render(<div className="todoList" todos={todosStub} />)
    expect(rendered).toMatchSnapshot();
  });

  it('should render correctly in "debug" mode', () => {
    const component = shallow(<TodosList debug todos={todosStub} />)
    expect(component).toMatchSnapshot();
  });

  it('task class should exist', () => {;
    expect(wrapper.find(".todoList").length).toEqual(1);
  });

  it('task class should exist', () => {;
    expect(wrapper.find(".todoItem").length).toEqual(4);
  });

  it('task class should exist', () => {;
    expect(wrapper.find(".todoItem").first().text()).toBe("1: delectus aut autem");
  });

});