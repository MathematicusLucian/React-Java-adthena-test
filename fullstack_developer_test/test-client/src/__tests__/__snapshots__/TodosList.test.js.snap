// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodosList should render (snapshot) 1`] = `
<div
  class="todoList"
  todos="[object Object],[object Object],[object Object],[object Object]"
/>
`;

exports[`TodosList should render correctly in "debug" mode 1`] = `
<div
  className="todoList"
>
  <div
    className="todoItem"
    key="1"
  >
    1
    : 
    delectus aut autem
  </div>
  <div
    className="todoItem"
    key="2"
  >
    2
    : 
    quis ut nam facilis et officia qui
  </div>
  <div
    className="todoItem"
    key="3"
  >
    3
    : 
    fugiat veniam minus
  </div>
  <div
    className="todoItem"
    key="4"
  >
    4
    : 
    et porro tempora
  </div>
</div>
`;
